doc/troubleshooting.md

```diff
@@ -1,5 +1,36 @@
 # Troubleshooting
 
+## How email sending is triggered by content changes
+
+We attempt to send emails to subscribers when content changes if they have a subscription for a subscribable that matches.
+
+A change to a content item creates a `ContentChange` record, which has a `SubscriptionContent` that belongs to an `Email`. The `SubscriptionContent` is linked to a `Subscription`, which has a `Subscriber`, and belongs to a `Subscribable`. A `DeliveryAttempt` is made on the `Email`.
+
+![domain](https://github.com/alphagov/email-alert-api/blob/master/doc/domain.png?raw=true)
+
+## The digest run workflow
+
+A daily task (for the daily digest) or a Monday task (for the weekly digest) runs as a scheduled task in the `DigestInitiatorService`. This creates a `DigestRun`, then fetches the IDs of the subscribers and creates a `DigestRunSubscriber` record for each one, before enqueuing the digest run subscribers in the `DigestGenerationWorker`.
+
+This worker uses a `SubscriptionContentChangeQuery` to fetch subscriptions with content changes, and creates an `Email` using the `DigestEmailBuilder` and composed of `SubscriptionContent`.
+
+![digests](https://github.com/alphagov/email-alert-api/blob/master/doc/digests.png?raw=true)
+
+## How Email Alert API interacts with Notify
+
+email-alert-service makes a POST request which is handled by the email-alert-api `ContentChangeController`
+to create a `ContentChange` and enqueue a `ProcessContentChangeWorker`.
+
+This Sidekiq worker uses `SubscriptionMatcher` to find the affected `SubscriptionContent` and enqueues an `EmailGenerationWorker`.
+
+The [digest run workflow](#the-digest-run-workflow) is followed in another Sidekiq worker to enqueue the email.
+
+This last Sidekiq worker creates a `DeliveryAttempt` and sends the email to GOV.UK Notify. It changes its status to `sent_to_notify`.
+
+A continuous process searches all delivery attempts for the `sent_to_notify` status and queries for a status update from Notify. If Notify reports a success, the `DeliveryAttempt` is set to `success`, otherwise it is given a fail state and an error message is logged on the record. At this point, logic plays out to, for example, retry in an hour, or blacklist the subscriber.
```

Comment:
This isn't quite right. Rather than us querying Notify, we wait for them to tell us about the status of emails using callbacks. There's more information about that process in their documentation: https://docs.notifications.service.gov.uk/ruby.html#delivery-receipts

Reply:
Thanks for the details! Looks like the diagram was out of date - only reference to DeliveryMonitor I could find was in ADR 1. Have now updated the diagram and the explanation here.